src/applications/common/schemaform/save-in-progress/actions.js

```diff
@@ -196,23 +204,32 @@ export function saveAndRedirectToReturnUrl(...args) {
  *                                version of the form the data was saved with
  *                                is different from the current version.
  */
-export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null) {
+export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null, prestartForm = null) {
+
   // TODO: Migrations currently aren’t sent; they’re taken from `form` in the
   //  redux store, but form.migrations doesn’t exist (nor should it, really)
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
+    if (prestartForm) {
+      try {
+        await prestartForm();
+      // If prestart error, fail safely
+      } catch (err) {
+        return;
+      }
+    }
     const trackingPrefix = getState().form.trackingPrefix;
     const userToken = sessionStorage.userToken;
     // If we don’t have a userToken, fail safely
     if (!userToken) {
       dispatch(setFetchFormStatus(LOAD_STATUSES.noAuth));
-      return Promise.resolve();
+      Promise.resolve();
     }
 
     // Update UI while we’re waiting for the API
     dispatch(setFetchFormPending(prefill));
```

Comment:
I believe `dispatch` is synchronous. I don't think this will cause any issues but it's probably better to finish the work in this function first (like the `fetch`) and `dispatch` at the end.

Reply:
Thanks @rianfowler, that's a great point, in this case, a user is not logged in, so we use Promise.resolve and exit the call without fetching the form.

Reply:
In fact, as async functions always return a promise, we should convert these resolve/rejects to returns and throws. Will update.